modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp

```diff
@@ -1464,18 +1481,576 @@ void run_sepfilter3x3_impl(DST out[], const SRC *in[], int width, int chan,  \
+        {
+            //  previous , this , next pixels
+            s[i] = in[k] + (i - border)*shift;
+        }
+
+        // rely on compiler vectoring
+        for (int l = 0; l < length; ++l)
+        {
+            float sum = 0;
+            for (int i = 0; i < kxLen; ++i)
+            {
+                sum += s[i][l] * kx[i];
+            }
+            buf[r[k]][l] = sum;
+        }
+    }
+
+    // vertical pass
+
+    for (int l = 0; l < length; ++l)
+    {
+        float sum = 0;
+        for (int j = 0; j < kyLen; ++j)
+        {
+            sum += buf[r[j]][l] * ky[j];
+        }
+
+        if (!noscale)
+        {
+            sum = sum * scale + delta;
+        }
+
+        out[l] = saturate<DST>(sum, rintf);
+    }
+    return;
+}
+
+template<bool noscale, typename DST, typename SRC>
+static void run_sepfilter5x5_code(DST out[], const SRC *in[], int width, int chan,
+                                  const float kx[], const float ky[], int border,
+                                  float scale, float delta, float *buf[], int y, int y0)
+{
+#if CV_SIMD
+    int length = width * chan;
+
+    // length variable may be unused if types do not match at 'if' statements below
+    (void)length;
+
+    if (std::is_same<DST, short>::value && std::is_same<SRC, uchar>::value &&
+        length >= v_int16::nlanes)
+    {
+        run_sepfilter5x5_char2short<noscale>(reinterpret_cast<short*>(out),
+                                             reinterpret_cast<const uchar**>(in),
+                                             width, chan, kx, ky, border, scale, delta,
+                                             buf, y, y0);
+        return;
+    }
+
+    if (std::is_same<DST, float>::value && std::is_same<SRC, float>::value &&
+        length >= v_float32::nlanes)
+    {
+        run_sepfilter5x5_any2float<noscale>(reinterpret_cast<float*>(out), in,
+            width, chan, kx, ky, border, scale, delta);
+        return;
+    }
+
+    if (std::is_same<DST, short>::value && length >= v_int16::nlanes)
+    {
+        run_sepfilter5x5_any2short<noscale>(reinterpret_cast<short*>(out), in,
+            width, chan, kx, ky, border, scale, delta,
+            buf, y, y0);
+        return;
+    }
+
+    if (std::is_same<DST, ushort>::value && length >= v_uint16::nlanes)
+    {
+        run_sepfilter5x5_any2short<noscale>(reinterpret_cast<ushort*>(out), in,
+            width, chan, kx, ky, border, scale, delta,
+            buf, y, y0);
+        return;
+    }
+
+    if (std::is_same<DST, uchar>::value && length >= v_uint8::nlanes)
+    {
+        run_sepfilter5x5_any2char<noscale>(reinterpret_cast<uchar*>(out), in,
+            width, chan, kx, ky, border, scale, delta,
+            buf, y, y0);
+        return;
+    }
+#endif  // CV_SIMD
+
+    // reference code is quite fast for any-to-float case,
+    // but not for any-to-integral due to very slow rounding
+    run_sepfilter5x5_reference<noscale>(out, in, width, chan, kx, ky, border,
```

Comment:
All this dispatcher logic can be unified for any kernel size if you add template parameter for kernel size. Moreover, I believe that almost all the code for 3x3 and 5x5 can be unified with such approach

Reply:
I will create a separate task for this according to the Dmitry's request.

Reply:
Ok